### ISSUE TITLE
Installer doesn't properly handle paths for Mason tooling

### ISSUE THREAD

**Issue by seantrons** [seantronsen/neovim-config#29]
The current iteration of the installer fails to account for mason tooling paths as of 4563cd8.

As a result, all mason tools and their associated directories are transported to the target installation path with invalid symlinks, particularly in the mason/bin directory. However, the issue also occurs wherever mason deems it necessary to create an external bash file to launch tools. 

In both cases, the path to the build on the original build system is specified, meaning that all affected tools will fail to run on different target systems.

**Comment by seantronsen** [comment]
Upgrading to mason [v2.0.0](https://github.com/mason-org/mason.nvim/releases/tag/v2.0.0) fixes **half** of this problem.

**Comment by seantronsen** [comment]
The other half, being `mason`'s use of a shell script to launch specific tooling, still exists. Inside those files, the absolute path to the script is always used.

**Comment by seantronsen** [comment]
I'll have to review it on a case by case basis, but we can probably hack a solution together by simply doing a find and replace search where: 
- search string is what "$HOME" expands to
- replace string is $HOME (not the expansion)

**Comment by seantronsen** [comment]
Of course, this also neglects to consider properties of bytecode and binary files. Depending on how `mason.nvim` handles those, each could be permanently fucked...
